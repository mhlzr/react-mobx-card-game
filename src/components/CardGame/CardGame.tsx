import React, { ReactElement, FunctionComponent, useEffect, useState } from 'react';
import styled from 'styled-components';
import { Provider } from 'mobx-react';

import { CardDetails } from '../CardDetails/CardDetails';
import { CardControls } from '../CardControls/CardControls';
import { CardOverview } from '../CardOverview/CardOverview';

import cardGameStore from '../../stores/CardGameStore';

const StyledDiv = styled.div`
    display: grid;
    grid-gap: 1em;
    grid-template-columns: 2fr 1fr;
    grid-template-areas:
        "details controls"
        "overview .";
    padding: 1em;
    width: 100vw;
`;

export const CardGame: FunctionComponent = (): ReactElement => {

    cardGameStore.fetchPlayers();

    return (
        <Provider store={cardGameStore}>
            <StyledDiv>
                <CardDetails />
                <CardControls />
                <CardOverview />
            </StyledDiv>
        </Provider>
    )
}
