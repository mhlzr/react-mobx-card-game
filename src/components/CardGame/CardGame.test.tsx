import React from 'react';
import { shallow } from 'enzyme'

import { CardGame } from './CardGame';

describe('CardGame', () => {
    it('has a snapshot', () => {
        const wrapper = shallow(<CardGame />);
        expect(wrapper).toMatchSnapshot();
    });
})
