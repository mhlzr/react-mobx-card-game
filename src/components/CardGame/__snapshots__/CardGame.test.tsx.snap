// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGame has a snapshot 1`] = `
<Provider
  store={
    CardGameStore {
      "_players": Array [],
      "fetchPlayers": [Function],
      "isFetching": true,
      "isSaving": false,
      "player": null,
      "playerSortation": null,
      "savePlayerSelection": [Function],
    }
  }
>
  <inject-_class-with-store />
</Provider>
`;
